test/rulesets/Base/broadcast.jl

```diff
@@ -179,5 +179,16 @@ BT1 = Broadcast.BroadcastStyle(Tuple)
     @testset "bugs" begin
         @test ChainRules.unbroadcast((1, 2, [3]), [4, 5, [6]]) isa Tangent   # earlier, NTuple demanded same type
         @test ChainRules.unbroadcast(broadcasted(-, (1, 2), 3), (4, 5)) == (4, 5)  # earlier, called ndims(::Tuple)
+
+        x = Base.Fix1.(*, 1:3.0)
+        dx1 = [Tangent{Base.Fix1}(; x = i/2) for i in 1:3, _ in 1:1]
+        @test size(ChainRules.unbroadcast(x, dx1)) == size(x)
+        dx2 = [Tangent{Base.Fix1}(; x = i/j) for i in 1:3, j in 1:4]
+        @test size(ChainRules.unbroadcast(x, dx2)) == size(x)  # was an error, convert(::ZeroTangent, ::Tangent)
+        # sum(dx2; dims=2) isa Matrix{Union{ZeroTangent, Tangent{Base.Fix1...}}, ProjectTo copies this so that
+        # unbroadcast(x, dx2) isa Vector{Tangent{...}}, that's probably not ideal.
+
+        @test sum(dx2; dims=2)[end] == Tangent{Base.Fix1}(x = 6.25,)
+        @test sum(dx1) isa Tangent{Base.Fix1}  # no special code required
```

Comment:
```suggestion
    # https://github.com/JuliaDiff/ChainRules.jl/issues/708
    @testset "unboardcast matrix"
            x = Base.Fix1.(*, 1:3.0)
            dx1 = [Tangent{Base.Fix1}(; x = i/2) for i in 1:3, _ in 1:1]
            @test size(ChainRules.unbroadcast(x, dx1)) == size(x)
            dx2 = [Tangent{Base.Fix1}(; x = i/j) for i in 1:3, j in 1:4]
            @test size(ChainRules.unbroadcast(x, dx2)) == size(x)  # was an error, convert(::ZeroTangent, ::Tangent)
            # sum(dx2; dims=2) isa Matrix{Union{ZeroTangent, Tangent{Base.Fix1...}}, ProjectTo copies this so that
            # unbroadcast(x, dx2) isa Vector{Tangent{...}}, that's probably not ideal.

            @test sum(dx2; dims=2)[end] == Tangent{Base.Fix1}(x = 6.25,)
            @test sum(dx1) isa Tangent{Base.Fix1}  # no special code required
    end
```